### PR TITLE
UserService methods tests

### PR DESCRIPTION
**Pull request by briocky**

### CODE REVIEW

**Review by aidian3k — CHANGES_REQUESTED** [review]
Changes needed as we talked

File: src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+  @BeforeEach
+  void openMocks() {
+    closeable = MockitoAnnotations.openMocks(this);
+  }
+
+  @AfterEach
+  void releaseMocks() throws Exception {
+    closeable.close();
+  }
```
Comment (aidian3k) [inline src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java:39]: what?

File: src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+import project.eepw.softwaretestingcrud.domain.user.entity.User;
+import project.eepw.softwaretestingcrud.infrastructure.exception.UserNotFoundException;
+
+class UserServiceTest {
+
+  @Mock
+  private UserRepository userRepository;
+  @InjectMocks
+  private UserService userService;
+  private AutoCloseable closeable;
```
Comment (aidian3k) [inline src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java:29]: ?

File: src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    //given
+    long id = 1L;
+    User user = User.builder()
+        .id(id)
+        .name("user1")
+        .email("user@example.com")
+        .surname("Doe")
+        .password("password")
+        .posts(Collections.emptySet())
+        .build();
```
Comment (aidian3k) [inline src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java:52]: extract all of these to some private static method and return only builder to set ids

File: src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+  }
+
+  @Test
+  void shouldThrowExceptionWhenUserWithThatIdIsNotInDB() {
+    //given
+    long notExistingUserId = 1L;
+    when(userRepository.findById(notExistingUserId)).thenReturn(Optional.empty());
+
+    //when
+    ThrowingCallable getUserByIdExecutable = () -> userService.getUserById(notExistingUserId);
```
Comment (aidian3k) [inline src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java:73]: It is not actually a callable but fine

File: src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+
+    //then
+    assertThatThrownBy(getUserByIdExecutable)
+        .isInstanceOf(UserNotFoundException.class)
+        .hasMessageContaining("User has not been found");
+    verify(userRepository).findById(notExistingUserId);
```
Comment (aidian3k) [inline src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java:79]: same here, what do we verify?

**Review by aidian3k — COMMENTED** [review]
Better, but you should still make some changes

File: src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java (hunk 1 above)
Comment (briocky) [reply src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java]: Done

**Review by aidian3k — APPROVED** [review]

**Pull request merged**